### PR TITLE
Fix mypy errors related to NetworkTimeout 

### PR DESCRIPTION
**Pull request by palango**
Fix mypy errors related to NetworkTimeout and Nonce

Part of #3798

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,6 +64,8 @@ mypy:
 	grep ChannelID mypy-out.txt; [ $$? -eq 1 ]
 	grep BalanceProof mypy-out.txt; [ $$? -eq 1 ]
 	grep SendSecret mypy-out.txt; [ $$? -eq 1 ]
```
Comment (LefterisJP) [inline Makefile]: just a note Paul. I think I am not opting for this kind of fixes. I am now taking each file and fixing it on its own so that we can soon just run mypy on the whole thing.

File: constraints-dev.txt [hunk 2]
```diff
@@ -85,7 +85,7 @@ sphinxcontrib-images==0.7.0
 sphinxcontrib-websupport==1.1.0
 treq==18.6.0
 Twisted==18.9.0
-typed-ast==1.1.2
+typed-ast==1.3.1
```
Comment (LefterisJP) [inline constraints-dev.txt]: why the change? What is it required for?

Comment (palango) [reply constraints-dev.txt]: This was updated when I upgraded mypy itself, so I guess it's a dependency of mypy.

File: Makefile (hunk 1 above)
Comment (palango) [reply Makefile]: The advantage of doing this is that it prevents regressions, but not all fixes are easily checkable by this.

Comment (LefterisJP) [reply Makefile]: yeah what I mean is we planned to just finish this issue kind of soonish. We started with this method and kind of stopped. So I want to just do it now and be done with it, as we decided to take these 2 weeks for the "tech debt" stuff.

We want raiden to become like the services and just run mypy on the whole codebase asap. So I am planning to make many small PRs like this one or the one you approved from me a few mins ago.

File: constraints-dev.txt (hunk 2 above)
Comment (LefterisJP) [reply constraints-dev.txt]: I see. Makes sense :+1:
